### DISCUSSION TITLE
Adding additional fields for insert/update using LinkingSetRelation attribute

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#6970] [Help]
Please I am implementing `LinkingSetRelation` because of `M-N mapping` scenario
Its working, however the  `N` table has additional columns that also need to be populated during the `Create/Update`.

How do I achieve this because it seems its only the `Foreign key` field and the `list values` field in the linking row that are considered.

``` c#
[DisplayName("Tenants"), Column("Tenants"), QuickFilterOption("multiple", true),Required,NotMapped]
[LookupEditor(typeof(TenantRow),Multiple =true), LookupInclude]
[LinkingSetRelation(typeof(ItemDynamicRow), "ProductId", "TenantId")]
public List<Int32> Tenants
{
    get { return Fields.Tenants[this]; }
    set { Fields.Tenants[this] = value; }
}
```

**Comment by rhemasystems** [comment]
Hi, Please any assistance would be greatly appreciated

Thank you

**Comment by rhemasystems** [comment]
Hi, Please any assistance would be greatly appreciated @VictorTomaili and @volkanceylan 

Thank you

**Comment by VictorTomaili** [comment]
LinkingSetRelation behavior inserts the values by sql query. Not by handlers. So, the insert/update fields are not filled automatically. If you need to fill them, you need to write a custom LinkingSetRelation behavior.

**Reply by rhemasystems** [reply]
Thank you @VictorTomaili , but how do I handle that,so I can write my own custom query, am new to it. Thank you

**Comment by rhemasystems** [comment]
@VictorTomaili , please any help from you will be greatly appreciated.
Thank you

**Comment by rhemasystems** [comment]
@VictorTomaili , please any help from you will be greatly appreciated, I cant seem to find any information in the documentation
Thank you

**Reply by VictorTomaili** [reply]
https://github.com/serenity-is/Serenity/blob/master/src/Serenity.Net.Services/RequestHandlers/IntegratedFeatures/LinkingSetRelation/LinkingSetRelationBehavior.cs

You just need to write the own behavior. Check the original one.

**Comment by volkanceylan** [comment]
Use MasterDetailRelation, not LinkingSetRelation

**Reply by rhemasystems** [reply]
Thank you @volkanceylan ,
Please thank you, am using that in some other areas of my project,just that the feature I want to use now as seen below, it seems its the  `LinkingSetRelation`  that will make it work, but please correct me if am wrong.

![image](https://github.com/serenity-is/Serenity/assets/50826372/e76c5ac6-83ed-429d-b541-bde827bf6350)

The table that stores the circled information has other mandatory fields not just the masterid and the itemkey

![image](https://github.com/serenity-is/Serenity/assets/50826372/6f2a3797-3919-451e-bd1f-a1faae3954f3)

**Reply by volkanceylan** [reply]
Yes, it is MasterDetail and a grid as usual, like OrderDetail
